src/addons/Radio/Radio.js

```diff
@@ -24,10 +27,22 @@ Radio.propTypes = {
     PropTypes.string,
     PropTypes.func,
   ]),
+
+  /** HTML input type, either checkbox or radio. */
+  inputType: Checkbox.propTypes.inputType,
+
+  /**
+   * Display as a checkbox, radio, slider, or toggle.
+   * The input type is `checkbox` for both slider and toggle types.
+   * You can set `inputType` separately to mix and match appearance and behavior.
+   */
+  type: Checkbox.propTypes.type,
 }
 
 Radio.defaultProps = {
   as: Checkbox,
+  inputType: 'radio',
```

Comment:
Though this PR does not refactor the Radio API, I agree completely on this.  I struggled with the naming there.

A SUI Checkbox can have an input type (radio|checkbox) and one of four "Checkbox types" (checkbox|radio|slider|toggle). The confusing thing to me about the current Checkbox API is the `type` prop, is it the input type or the SUI Checkbox type?

This is what spurred the Radio component, separation of input `type`.  We still have confusion of the "SUI Checkbox type".  Now that we a Radio, maybe we remove the ability to set the input type at all.  Intuition says the input type is already set by the fact that it is a checkbox or radio component imo.

If so, what then should we call the SUI type?  I think `style` or `type` are best suited for this prop's purpose, but both of those are seriously conflicting prop names.  It could be `appearance`, `kind`, ...something better:

```jsx
// always input type radio
<Radio />
<Radio appearance='slider' />
<Radio appearance='toggle' />

// always input type checkbox
<Checkbox />
<Checkbox appearance='slider' />
<Checkbox appearance='toggle' />

// strange edge cases, if you must
<Radio appearance='checkbox' />
<Checkbox appearance='radio' />
```

Another option is to create a component for each visual type, and accept a `type` prop only for the input type.  IMO this is more intuitive regarding input type, but seems to unnecessarily fracture the top-level API where a prop could have handled the styling differences.

```jsx
<Slider />                      // checkbox default
<Slider type='radio' />

<Toggle />                      // checkbox default
<Toggle type='radio' />

// strange edge cases, if you must
<Radio type='checkbox' />
<Checkbox type='radio' />
```